apps/framework-cli/src/framework/schema.rs

```diff
@@ -376,10 +384,154 @@ pub fn ast_mapper(ast: SchemaAst) -> Result<FileObjects, ParsingError> {
     Ok(FileObjects::new(parsed_models, enums))
 }
 
+pub fn ts_ast_mapper(ast: Module) -> Result<FileObjects, ParsingError> {
+    let models = Vec::new();
+    let mut enums = Vec::new();
+
+    let mut ts_declarations = Vec::new();
+
+    // collect all inteface and enum declarations
+    ast.body.iter().for_each(|item| match item {
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsInterface(decl))) => {
+            ts_declarations.push(decl);
+        }
+        ModuleItem::Stmt(Stmt::Decl(Decl::TsEnum(decl))) => {
+            enums.push(ts_enum_to_data_enum(decl));
+        }
+        _ => {}
+    });
+
+    println!(
+        "the interfaces {:#?}",
+        ts_interface_to_model(ts_declarations[0], &enums)
+    );
+
+    Ok(FileObjects::new(models, enums))
+}
+
+fn ts_enum_to_data_enum(enum_decl: &TsEnumDecl) -> DataEnum {
+    let name = enum_decl.id.sym.to_string();
+    let values = enum_decl
+        .members
+        .iter()
+        .map(|member| match member {
+            TsEnumMember {
+                id: TsEnumMemberId::Ident(ident),
+                ..
+            } => ident.sym.to_string(),
+            _ => "unsupported".to_string(),
```

Comment:
```
.map(|member| match &member.id {
    TsEnumMemberId::Ident(ident) => ident.sym.to_string(),
    TsEnumMemberId::Str(str) => str.value.to_string()
})
```

Reply:
If enum values are numbers, then we should use those numbers in clickhouse as well.

Reply:
If enum values are strings, then we should use the values instead of the identifiers, otherwise we'll have to do some mapping in the send functions before calling `JSON.stringify`.

```
enum Direction {
  Up = "UP",
  Down = "DOWN",
  Left = "LEFT",
  Right = "RIGHT",
}
```

And if the enum values are other stuff, we should probably reject the schema.